chain-impl-mockchain/src/vote/manager.rs

```diff
@@ -158,6 +165,49 @@ impl ProposalManager {
         })
     }
 
+    #[must_use = "Compute the PrivateTally in a new ProposalManager, does not modify self"]
+    pub fn private_tally<F>(
+        &self,
+        stake: &StakeControl,
+        governance: &Governance,
+        f: &mut F,
+    ) -> Result<Self, VoteError>
+    where
+        F: FnMut(&VoteAction),
+    {
+        let mut tally = chain_vote::Tally::new(self.options.as_byte() as usize);
+
+        for (id, payload) in self.votes_by_voters.iter() {
+            if let Some(account_id) = id.to_single_account() {
+                if let Some(stake) = stake.by(&account_id) {
+                    match payload {
+                        vote::Payload::Public { .. } => {
+                            return Err(VoteError::InvalidPayloadType {
+                                expected: vote::PayloadType::Private,
+                                received: vote::PayloadType::Public,
+                            });
+                        }
+                        vote::Payload::Private { encrypted_vote } => {
+                            tally.add(encrypted_vote, stake.0);
+                        }
+                    }
+                }
+            }
+        }
+
+        // TODO: What to check here?
+        // if self.check(stake.assigned(), governance, &results) {
+        //     f(&self.action)
+        // }
```

Comment:
I don't really know what should I do here

Reply:
So, as far as I understand, this is used for the on-chain "automatic" tally, which I'm not sure if we need for the moment. But, the way I see it, I don't think having this function makes sense, as it can't be executed by non-committee nodes. I imagine that for automatic on-chain actions with private votes, the committee members would need to publish a new fragment type with the `TallyDecryptShare` in order for all the nodes to validate the tally, and take governance actions.
(If this is the case I think it would be better to have two different types, btw)